src/Daemon/Daemon.cpp

```diff
@@ -460,6 +466,44 @@ namespace usbguard
   {
   }
 
+  pid_t Daemon::pid_fd = -1;
+
+  void Daemon::daemonize(const std::string &pid_file)
+  {
+      pid_t pid = 0;
+
+      USBGUARD_SYSCALL_THROW("Daemonize", (pid = fork()) < 0);
+      if (pid > 0) {
+          exit(EXIT_SUCCESS);
+      }
+
+      /* Now we are forked */
+      USBGUARD_SYSCALL_THROW("Daemonize", setsid() < 0);
+      signal(SIGCHLD, SIG_IGN);
+
+
+      USBGUARD_SYSCALL_THROW("Daemonize", (pid = fork()) < 0);
+      if (pid > 0) {
+          exit(EXIT_SUCCESS);
+      }
+
+      /* Now we are forked 2nd time */
+      umask(0047);  /* no need for world-accessible or executable files */ 
+      chdir("/");
+      /* We do not need to close all fds because there is only logging open at this point */
+      stdin = fopen("/dev/null", "r");
```

Comment:
Shouldn't the stdin/err/out redirecting be done on the file descriptor level? Where do you take care of closing 0,1,2 fds? I think we should use the standard close+open+dup2 process here.